playbook-website/app/javascript/components/Website/src/hooks/RedirectToType.tsx

```diff
@@ -0,0 +1,16 @@
+import React from "react"
+import { useParams, useSearchParams, Navigate } from "react-router-dom"
+
+export default function RedirectToType() {
+  const { name } = useParams()
+  const [searchParams] = useSearchParams()
+  const queryType = searchParams.get("type")
+
+  //  Supports our old routing, keeping our routes backwards-compatible.
+  if (queryType) {
+    return <Navigate to={`/beta/kit_category/${name}/${queryType}`} replace />
+  }
+
+  // If there's no queryType, navigate to the defaultType (react).
+  return <Navigate to={`/beta/kit_category/${name}/react`} replace />
```

Comment:
I think that, instead, you should dynamically build the path and only return one instance of `<Navigate...`

Reply:
Cool. I like it. I'll do that. This scaffold is really a future view anyways, and isn't being utilized yet for my current story--- but i'll go ahead and make that change as well.